client/components/happychat/button.jsx

```diff
@@ -1,42 +1,92 @@
 /**
  * External dependencies
  */
-import React, { Component } from 'react';
+import React, { Component, PropTypes } from 'react';
 import page from 'page';
+import { identity, noop } from 'lodash';
 import { connect } from 'react-redux';
 import { localize } from 'i18n-calypso';
 import Gridicon from 'gridicons';
+import classnames from 'classnames';
 
 /**
  * Internal dependencies
  */
 import viewport from 'lib/viewport';
+import {
+	isHappychatAvailable,
+	isHappychatChatActive,
+} from 'state/happychat/selectors';
+import { connectChat } from 'state/happychat/actions';
 import { openChat } from 'state/ui/happychat/actions';
 import Button from 'components/button';
 
 class HappychatButton extends Component {
-	onOpenChat = () => {
-		const { onOpenChat } = this.props;
+	static propTypes = {
+		borderless: PropTypes.bool,
+		connectChat: PropTypes.func,
+		isChatActive: PropTypes.bool,
+		isChatAvailable: PropTypes.bool,
+		onClick: PropTypes.func,
+		openChat: PropTypes.func,
+		translate: PropTypes.func,
+	};
+
+	static defaultProps = {
+		borderless: true,
+		connectChat: noop,
+		isChatActive: false,
+		isChatAvailable: false,
+		onClick: noop,
+		openChat: noop,
+		translate: identity,
+	};
+
+	onClick = ( event ) => {
 		if ( viewport.isMobile() ) {
-			// For mobile clients, happychat will always use the page compoent instead of the sidebar
+			// For mobile clients, happychat will always use the
+			// page componet instead of the sidebar
 			page( '/me/chat' );
 		} else {
-			onOpenChat();
+			this.props.openChat();
 		}
+
+		this.props.onClick( event );
+	}
+
+	componentWillMount() {
+		this.props.connectChat();
 	}
 
 	render() {
-		const { translate } = this.props;
+		const { translate, children, className, borderless, isChatAvailable, isChatActive } = this.props;
+		const showButton = isChatAvailable || isChatActive;
+		const classes = classnames( 'happychat__button', className, {
+			'is-borderless': borderless,
```

Comment:
I mentioned this in my last review but the comment got collapsed through some changes:

What's the benefit of manually adding this class rather than passing `borderless` as a prop to `<Button>`?